posthog/migrations/0039_populate_event_ip_property.py

```diff
@@ -0,0 +1,23 @@
+from django.db import migrations
+
+def migrate_event_ip_to_property(apps, schema_editor):
+    Event = apps.get_model('posthog', 'Event')
+    events = Event.objects.all()
```

Comment:
I think this might cause a MemoryError because all the events are being read from the database into memory. @timgl mentioned that it should be able to handle a couple of million rows.

A better approach, in this case, would be to bulk update 10000 events at a time using Django's [bulk_update](https://docs.djangoproject.com/en/3.0/ref/models/querysets/). It's also a lot faster than having to update each model at a time since it will only do query per batch.

Reply:
cool, will give this a shot test for couple of million of rows.

Reply:
Good thinking. I do wonder if it would be way better to just do this as a raw SQL query, rather than loading any of it inside Python. That being said, if your testing does work I'm happy with a python only solution

Reply:
I implemented the bulk_update, referring to the Django documentation. Also changed the test case to use bulk_create to efficiently create 2 millions records. The test cases ran fine even with 2 million records.

@gzog @timgl

Reply:
This approach does not solve loading all events into memory. I think no matter what, if we are using bulk update, we will have to use Event.objects.all() at some point to ensure that during batch processing same order of event objects is preserved while preparing the batch. Let me know if I am missing anything?

Reply:
I think you can use [Queryset.iterator](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#iterator) to batch the reads as well.

You can do Event.objects.all().iterator(chunk_size=10000) to just read 10k objects at a time into memory, so that's how you keep the memory footprint low and then do bulk_update on those 10k objects so that the writes to the DB are fast.

See https://stackoverflow.com/a/31525594/2113279 for more info on this.

Reply:
Ok, so I handled loading all objects into memory by using the [iterator](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#iterator). This hopefully ought to do it.

Reply:
A nice. I just noticed that you made the update. good job 👍 @bhavish-agarwal